test/src/kernels/PhiZeroKernel.C

```diff
@@ -0,0 +1,70 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PhiZeroKernel.h"
+
+registerMooseObject("MooseTestApp", PhiZeroKernel);
+
+InputParameters
+PhiZeroKernel::validParams()
+{
+  InputParameters params = NullKernel::validParams();
+  params.addClassDescription("Test for phi zero values");
+  params.addParam<bool>("second_order", false, "use second order shape function");
+  return params;
+}
+
+PhiZeroKernel::PhiZeroKernel(const InputParameters & parameters)
+  : NullKernel(parameters), _second_u(_var.secondSln()), _second_phi(_var.secondPhi())
+{
+}
+
+Real
+PhiZeroKernel::computeQpResidual()
+{
+
+  if (_phi.size() > 0)
+  {
+    mooseAssert(_phi.size() == _phi_zero.size(),
+                "_phi.size() " + std::to_string(_phi.size()) + "> _phi_zero.size() " +
+                    std::to_string(_phi_zero.size()));
```

Comment:
I understand that you were thinking about cases in which the "zero" members were of zero size, but your message should probably just say "!=" as opposed to ">". Same for all the following assertions